litestar/middleware/compression/facade.py

```diff
@@ -29,11 +29,13 @@ def __init__(
         """
         ...
 
-    def write(self, body: bytes) -> None:
-        """Write compressed bytes.
+    def write(self, body: bytes | bytearray, final: bool = False) -> None:
```

Comment:
Changing this to `bytearray` and introducing the `final` parameter makes this a breaking change. That's okay, it just means we can't release this in the `2.x` release line and it'll have to wait until `3.0`